internal/pikamon/bot.go

```diff
@@ -27,14 +27,17 @@ func New(cfg *Config) (*Bot, error) {
 		return nil, err
 	}
 
-	listener, err := spawn.NewHandler(cfg.Bot.SpawnChance, cfg.Bot.DebounceWindow)
+	// Create bot cache
+	createBotCache(cfg)
```

Comment:
So, I think the proper way to handle the cache issue is to have this return a pointer to the cache, attach it to the `Bot` struct, pass it into the spawn Handler, attach it to the `Handler` struct. Then, the commands stuff should be modified to also be a handler in the same fashion as the spawner. This way the cache can be attached the commands Handler struct, and accessible to it. Finally, the commands would become methods attached to the struct so the commands function has access to the cache.

I know this seems like a lot more verbosity, but Go prefers verbosity over implicitness.

Reply:
Created this issue which I will start working on next
- https://github.com/Jac0bDeal/pikamon/issues/24